tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/Flatten.scala

```diff
@@ -50,11 +50,14 @@ object Flatten {
     ex match {
       case LetInEx( body, defs@_* ) =>
         // Transform bodies of all op.defs
-        val newDefs = defs.map { x =>
-          x.copy(
-            body = self( x.body )
-          )
-        }
+        def xform: TlaOperDecl => TlaOperDecl =
+          tracker.trackOperDecl { d: TlaOperDecl =>
+            d.copy(
+              body = self(d.body)
+            )
+          }
+
+        val newDefs = defs map xform
```

Comment:
You could have just written `defs map (tracker.trackOperDecl { ... })`

Reply:
Agree. This is less verbose:  `defs map tracker.trackOperDecl { d => d.copy(body = self(d.body)) }`. I sometimes forget how easy it is to stack functions in Scala :)